vortex-serde/src/layouts/pruning.rs

```diff
@@ -86,7 +87,9 @@ impl<'a> PruningPredicateRewriter<'a> {
     ) -> Option<Self> {
         // TODO(robert): Simplify expression to guarantee that each column is not compared to itself
         //  For majority of cases self column references are likely not prunable
-        if other_exp.references().contains(&column) {
+        let mut refs = HashSet::new();
+        other_exp.collect_references(&mut refs);
```

Comment:
nit: consider adding a `references` fn to the trait that creates a HashSet and then calls collect_references so that we don't have to do this two-liner in multiple places?